src/borg/archiver/check_cmd.py

```diff
@@ -37,8 +37,6 @@ def do_check(self, args, repository):
             )
         if args.repair and args.max_duration:
             raise CommandError("--repair does not allow --max-duration argument.")
-        if args.undelete_archives and not args.repair:
```

Comment:
I **assume** that we also need the following here, right? If not, also remove the matching sentence in my docs suggestion below.

```py
if args.repo_only and args.find_lost_archives:
    raise CommandError("--repository-only contradicts the --find-lost-archives option.")
```

Reply:
ah, right! good catch!

Reply:
fixed.